### DISCUSSION TITLE
[@types/styled-components] Massive slow down in vscode still happening

### DISCUSSION THREAD

**Discussion by mkvlrn-cm42** [DefinitelyTyped/DefinitelyTyped#58425] [Issues with a @types package]
Adding the type definitions slow down VSCode to the point of being unusable.

[This discussion](https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/55970) was opened last year and [this issue](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/34391) from even before state the problem that persists to this day with the lates version of the definitions.

For the time being I'm using [@emotion/styled](https://github.com/emotion-js/emotion/tree/main/packages/styled), which works pretty much the same way. But I would like to use styled-components with working type definitions.

[a comment by an automated account is not shown]

**Comment by lifeiscontent** [comment]
@mkvlrn I've also experienced this, I've actually begun to rewrite the definitions from scratch in a personal project and have seen considerable speed improvements, though I've only typed the APIs I need so a lot of the definition is missing, if you are interested I can post my source code for the definition.
